takari-builder-common/src/test/java/io/takari/builder/internal/pathmatcher/PathNormalizerTest.java

```diff
@@ -0,0 +1,86 @@
+package io.takari.builder.internal.pathmatcher;
+
+import static io.takari.builder.internal.pathmatcher.PathNormalizer.normalize0;
+import static io.takari.builder.internal.pathmatcher.PathNormalizer.getCanonicalPath;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class PathNormalizerTest {
+
+  @Rule
+  public final TemporaryFolder temp = new TemporaryFolder();
+
+  @Test
+  public void testIsBasedirOrNestedFile() throws Exception {
+    Path basedir = temp.newFolder().getCanonicalFile().toPath();
+    PathNormalizer testee = PathNormalizer.create(basedir);
+
+    assertTrue(testee.isBasedirOrNestedFile(basedir.toString()));
+    assertTrue(testee.isBasedirOrNestedFile(basedir.resolve("nested").toString()));
+
+    assertFalse(testee.isBasedirOrNestedFile(temp.getRoot().getCanonicalPath()));
```

Comment:
Should you add a test that covers the following? based = "/basedir" -> testee.isBasedirOrNestedFile("/basedirxxx") // false

Reply:
Yes, right, meant to add such test but forgot.